### PR TITLE
7963 multiprocessing option for ES commands

### PR DESCRIPTION
**Pull request by aj-he**
Provided the capability to index resources using multiprocessing via the `es` management command. 

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Provided the capability to index resources using multiprocessing via the `es` management command. This is enabled by using the two new command parameters:

- `-mp` / `--use_multiprocessing`  flag to enable multiprocessing
-  `-mxp` / `--max_subprocesses`  (optional) - explicitly sets the maximum number of processes created by the process pool. Default = ceil(cpu_count / 2)

These parameters are available on the following es management module operations:

- `index_database`
- `reindex_database`
- `index_resources`

Default behaviour of these operations are unchanged. Using `--use_multiprocessing` enables the functionality.

**example usage**:
`$ python manage.py es index_database --use_multiprocessing`
`$ python manage.py es index_database -mp -mxp 6` where you want the process pool to explicitly make 6 processes available.

**Limitations**
A combination of batch size, and the number of processes being used can have adverse effects if the configuration of ElasticSearch means there are not enough resources to accept the load. Large batch sizes coupled with a lot of processes may cause ES to reject some index calls if it does not have enough memory.

As such, the use of `-mp` should remain an optional flag for developers/technical users.

### Issues Solved

#7963 #4858

### Checklist

-   [x] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: Historic England
*   Tested by: @aj-he @csmith-he 
*   Designed by: @aj-he

### Further comments

I will add the new options to arches-docs once the functionality has been reviewed and merged.

### CODE REVIEW

**Comment by aj-he** [comment]
@robgaston @chiatt reassign review as required. thx

**Comment by chiatt** [comment]
Thanks @aj-he, I'll have a look.

File: arches/app/search/search.py [hunk 1]
```diff
@@ -21,8 +21,9 @@
 import urllib.error
 import uuid
 import logging
+import warnings
 from datetime import datetime
-from elasticsearch import Elasticsearch, helpers
+from elasticsearch import Elasticsearch, helpers, ElasticsearchWarning
```
Comment (robgaston) [inline arches/app/search/search.py:26]: `ElasticsearchWarning` was added in `elasticsearch` [7.11.0](https://newreleases.io/project/pypi/elasticsearch/release/7.11.0), but our `requirements.txt` current specifies `elasticsearch>=7.11.0,<8.0.0`, so we should update this accordingly.

**Review by robgaston — CHANGES_REQUESTED** [review]
@aj-he these changes look good to me, as does my initial testing, but we should update `requirements.txt` in this PR as per my review comment.  Thanks!

File: arches/app/search/search.py (hunk 1 above)
Comment (aj-he) [reply arches/app/search/search.py:26]: I've updated requirements.txt to `elasticsearch>=7.11.0,<8.0.0` as per your request.

**Review by robgaston — APPROVED** [review]

**Pull request merged**